src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -23,6 +23,7 @@ export interface IFacetValueSuggestionsOptions {
   useQuerySuggestions?: boolean;
   useValueFromSearchbox?: boolean;
   displayEstimateNumberOfResults?: boolean;
+  expression?: string;
```

Comment:
Based on the jira, it seems that the filter expression is being set on a different component. Is there a way we can retrieve that filter expression and use it rather than requiring the implementer to configure the same filter expression in multiple places? Wouldn't that make configuring easier? 

That said, depending on where the initial expression is configured, the above may or may not make sense. If the `FacetValueSuggestions` component must be used in conjunction with the other component, then I think we should retrieve it. If the components are independent, then having a separate option makes more sense.

Reply:
We could also have a hybrid model where we use the option if it is specified. If it's not, we try to retrieve it.

Reply:
For what I've seen on the users's website, they are adding an expression to the batch values endpoint thru custom code. So I think it's more convenient to use a property on this component than having them to create code snippets.

Reply:
I think the custom code approach is a work-around to the option not being available, but also to the component not reading the other place where the filter is set. I am not certain, but from reading the jira:

> ... there is no way to add a filter in the request, so suggestions are misaligned with the content of the search page (**that have some filters on it**)

It would be good to know where the other filters on the search page are being configured. Maybe all the filters are being applied through custom code, but I have a feeling they are setting the filter using an exposed option on a different component, and the FacetValueSuggestions is not taking it into account.

Reply:
Ok. Will ask in the Jira.

Reply:
There's nowhere to "retrieve" the expression from in the context of a standalone search box. And that's what we're trying to solve here.

Say you had a full search page, with tabs. facets (or any other component that would add expression in the interface), as well as a standalone search box on the homepage of your website.

You need to replicate the filter that gets applied from the full search interface onto the FacetValueSuggestions component in the standalone search box, otherwise you might be able to select a facet value suggestions on the homepage, get redirected to the full search pages, and then get no results.

Both "pages" (full search page & standalone search box on homepage) won't have the same components/markup/configuration. 

In the context of the full search page, the FacetValueSuggestion component is already making sure it "copies" the last executed query, so all filters set through tabs/facets/any other custom code with filters will normally be taken into account.

The problem is really in the context of a standalone search box.

Reply:
Thanks, I was missing the context that this was in a stand-alone use-case. Just wanted to make sure we were not requiring developers to copy the same expression into multiple places unless necessary.